docs/virtual-env.md

```diff
@@ -2,8 +2,39 @@
 
 ## Dependencies
 
-- [Python 3](http://www.python.org/) (Can be installed using `brew install python3`)
-- [nodejs.org](http://nodejs.org/) (v10.24 - can be installed using [nvm](https://github.com/creationix/nvm))
+### Pyenv, python3
+
+"pyenv" is used to provide python3. (Other CLA repos need different versions, so good to be able to .)
+
+1. Install pyenv with brew:
+
+        brew install pyenv
+
+2. Set up your shell for pyenv. Make the changes to `~/.zshrc` described here: [Set up your shell for pyenv](https://github.com/pyenv/pyenv#set-up-your-shell-environment-for-pyenv) (This is so that pyenv's python binary can be found in your path)
+
+3. To make the shell changes take effect:
+
+        exec "$SHELL"
+
+    (or alternatively, restart your shell)
+
+4. Install into pyenv the python version this repo uses (which is defined in `.python-version`):
+
+        pyenv install 3.12 --skip-existing
+
```

Comment:
I would add this step. 

```
5. Select the 'Pyenv-installed' Python as the version to use, by running the following command:

`pyenv local 3.12`
```

Reply:
Good call about adding a version check. `pyenv local` will change `.python-version` though, which is probably not what's needed here. The stumbling block I've encountered is if you've created a venv in one python version, it's hard to switch the version, so I've added a paragraph on this. But if there is another pain point here, let me know